esmf_regrid/esmf_regridder.py

```diff
@@ -173,10 +175,10 @@ def _index_offset(self):
         return 1
 
     def _flatten_array(self, array):
-        return array.flatten()
+        return array.reshape(-1, (self.size()), order="F").T
```

Comment:
Perhaps `self.size` should be a `@property`? In any case, I think the `()` around `self.size()` don't do anything here, right?

I suggest to remove the superfluous `()` and consider making size a property, though perhaps in a different PR.

Reply:
Two more questions:

1. The transposition is here to allow for the application to be executed by matrix-vector multiplication *from the left* instead of from the right, correct?
2. Do we really need to specify the order and are we sure this does what we want? I was trying to read a bit more about `reshape`'s handling of the order parameter in [the documentation](https://numpy.org/doc/stable/reference/generated/numpy.reshape.html), but I can't say that it fills me with confidence. Note for example:
>  Note there is no guarantee of the memory layout (C- or Fortran- contiguous) of the returned array.

And have a look at the explanation of the order parameter in the Notes section with the examples.

If you are sure this is all right, I suggest adding a note either as a comment or in a docstring that explains why this is here and how it helps us.

Reply:
@stephenworsley let's talk this through today, I certainly agree with @zklaus that a comment here explaining why it's necessary.  In general, anywhere I put a `.T` is often a good signal that a comment is needed... I few weeks later often forget why it was the right thing to do !

Reply:
Great additional doc @stephenworsley 👍

Reply:
The `()` can still go away, right?